locksmith/src/websub/jobs/keys.ts

```diff
@@ -102,3 +122,233 @@ export async function notifyOfKeys(hooks: Hook[]) {
+      const keys = await subgraph.keys(
+        {
+          first: 1000, //  TODO: handle more than 1000 keys
+          where: {
+            expiration_gt: now.getTime().toString(),
+            expiration_lt: end.getTime().toString(),
+          },
+        },
+        {
+          networks: [Number(networkId)],
+        }
+      )
+
+      logger.info(
+        `keys expiring for ${networks[networkId].name}: ${keys?.length}`
+      )
+
+      // split keys into groups to send email by group with some delay
+      const chunks = arrayToChunks(keys, KEYS_CHUNKS_SIZE)
+      await Promise.allSettled(
+        Object.values(chunks).map(async (keys, index) => {
+          const timeout = DELAY * (index + 1) // increase delay for each chunks
+          await delay(timeout)
+
+          logger.info(
+            `send expiring emails for chunk group ${index + 1} on ${
+              networks?.[networkId]?.name
+            }`
+          )
+
+          await Promise.allSettled([
+            keys?.map(async (key: any) => {
+              const lockName = key?.lock?.name ?? ''
+              const lockAddress = Normalizer.ethereumAddress(key.lock.address)
+              const ownerAddress = Normalizer.ethereumAddress(key.owner)
+              const tokenAddress = key?.lock?.tokenAddress ?? ''
+              const keyId = key?.tokenId ?? ''
+
+              const protectedData =
+                await metadataOperations.getUserProtectedMetadata({
+                  lockAddress,
+                  userAddress: ownerAddress,
+                })
+
+              const recipient = protectedData?.email as string
+
+              if (!recipient) return
+
+              const {
+                isAutoRenewable,
+                isRenewable,
+                isRenewableIfRePurchased,
+                isRenewableIfReApproved,
+                currency,
+              } = await getMembershipStatus({
+                owner: ownerAddress,
+                key,
+                tokenAddress,
+                tokenId: keyId,
+                lockAddress,
+                network: Number(networkId),
+              })
+
+              // expiration date example: 1 December 2022 - 10:55
+              const expirationDate = dayjs(new Date(key.expiration)).format(
+                'D MMMM YYYY - HH:mm'
+              )
+
+              // send expiring email
+              await sendEmail(`keyExpiring`, `keyExpiring`, recipient, {
+                lockName,
+                keyId,
+                network: networkId,
+                keychainUrl: `${config.unlockApp}/keychain`,
+                currency,
+                expirationDate,
+                isRenewable: isRenewable ? 'true' : '',
+                isAutoRenewable: isAutoRenewable ? 'true' : '',
+                isRenewableIfRePurchased: isRenewableIfRePurchased
+                  ? 'true'
+                  : '',
+                isRenewableIfReApproved: isRenewableIfReApproved ? 'true' : '',
```

Comment:
```suggestion
                isRenewable: isRenewable ? 'true' : '',
                isAutoRenewable: isAutoRenewable ? 'true' : '',
                isRenewableIfRePurchased: isRenewableIfRePurchased
                  ? 'true'
                  : '',
                isRenewableIfReApproved: isRenewableIfReApproved ? 'true' : '',
```

I dont think we need to use boolean here, not accepted as valid value, if we look at `params` the typedefinition is `[key: string]: string | number | undefined`: 

also when values are set as `false` it will show in the template 

<img width="500" alt="Screenshot 2023-03-23 at 12 43 11" src="https://user-images.githubusercontent.com/20865711/227207248-e06782a0-59b7-49db-bf2a-1cab339e64cc.png">

we need to have an empty string or `undefined` to hide it

<img width="500" alt="Screenshot 2023-03-23 at 12 44 42" src="https://user-images.githubusercontent.com/20865711/227207634-a0837b40-80ae-4a6b-b3be-d7a4a452fdfa.png">

Reply:
> I dont think we need to use boolean here, not accepted as valid value, if we look at params the typedefinition is [key: string]: string | number | undefined:

Then we need to change the type :) It does not make sense to use `"false"` or `"true"` :)